exercises-cpp/mohammed_almakki/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include <string>
+
+class Shape
+{
+public:
+    virtual std::string getType() = 0;
+    virtual int getSidesCount() = 0;
+    virtual double getArea() = 0;
+    virtual double getPerimeter() = 0;
+
+    virtual std::string toString() = 0;
+
+    Shape(int noOfSides) : noOfSides{noOfSides} {}
+
+private:
+    int noOfSides;
+};
+
+class Square : public Shape
+{
+public:
+    Square(double side1) : side1{side1}, Shape(2) {}
+
+    std::string getType()
+    {
+        return "Square";
+    }
+
+    int getSidesCount()
```

Comment:
Could we have this defined in the `Shape` parent class, simply returning its member `noOfSides`? We then would not have to redefine it in every child.

Reply:
Yeah, that is really bad. Doesn't know why I did it like this.